src/main/java/seedu/address/model/Entity/Team.java

```diff
@@ -0,0 +1,234 @@
+package seedu.address.model.Entity;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class Team extends Entity {
+    private List<Participant> participants;
+    private Optional<Mentor> mentor;
+    private SubjectName subject;
+    private Score score;
+    private Name projectName;
+    private ProjectType projectType;
+    private Location location;
+
+    /**
+     * Constructor without mentor.
+     *
+     * @param teamId
+     * @param teamName
+     * @param participants
+     * @param subject
+     * @param score
+     * @param projectName
+     * @param projectType
+     * @param location
+     */
+    public Team(
+            Id teamId,
+            Name teamName,
+            List<Participant> participants,
+            SubjectName subject,
+            Score score,
+            Name projectName,
+            ProjectType projectType,
+            Location location
+    ) {
+        super(teamId, teamName);
+        this.participants = participants;
+        this.mentor = Optional.empty();
+        this.subject = subject;
+        this.score = score;
+        this.projectName = projectName;
+        this.projectType = projectType;
+        this.location = location;
+    }
+
+    /**
+     * Constructor with mentor.
+     *
+     * @param teamId
+     * @param teamName
+     * @param participants
+     * @param subject
+     * @param score
+     * @param projectName
+     * @param projectType
+     * @param location
+     */
+    public Team(
```

Comment:
Could you provide a getter method for teamName as well? I.e. getTeamName()

Reply:
That’s implemented in the parent abstract class, maybe I should change teamName to name to avoid confusion. :smile:

Reply:
getId() as well. Thanks!

Reply:
GetId also